### ISSUE TITLE
DROP TABLE doesn't work in saved Project

### ISSUE THREAD

**Issue by bartek990** [sqlitebrowser/sqlitebrowser#3315]
Details for the issue
--------------------
DROP TABLE command doesn't work in _Execute SQL_ with saved Projects.

#### What did you do?
I wrote and run a code:
```
CREATE TABLE IF NOT EXISTS stock (
	company		TEXT NOT NULL,
	stock_exchange	TEXT NOT NULL,
	price		REAL,
	PRIMARY KEY (company, stock_exchange)
);

INSERT INTO stock (company, stock_exchange, price)
VALUES	('ABC', 'GPW', 180.0),
	('DEF', 'GPW', 190.0);

DROP TABLE stock;
```
It was working perfectly, so I saved my database as Project.

On another day I loaded my project and ran it.

#### What did you expect to see?
That table stock disappeared from tree in DB Schema.

#### What did you see instead?
"Console":
```
Execution finished without errors.
```

SQL Log:
```
-- EXECUTING LINE IN '05_primaty_key.sql'
--
```
Screen:
![Zrzut ekranu z 2023-04-02 19-20-27](https://user-images.githubusercontent.com/6698414/229368648-bb266142-95a8-4555-b83c-5031aee893c6.png)

Terminal code to compare:
```
$ sqlite3
SQLite version 3.40.1 2022-12-28 14:03:47
Enter ".help" for usage hints.
Connected to a transient in-memory database.
Use ".open FILENAME" to reopen on a persistent database.
sqlite> .open app.db
sqlite> .tables
stock  user 
sqlite> DROP TABLE stock;
sqlite> .tables
user
sqlite> 
```

Useful extra information
-------------------------
> DB4S v3.12.99 (Mar 15 2023) [built for x86_64-little_endian-lp64] on Ubuntu Core 20 (linux/5.19.0-38-generic) [x86_64]
> using SQLCipher Version 4.4.3 community (based on SQLite 3.34.1)
> and Qt 5.12.8

**Comment by chrisjlocke** [comment]
Its weird it says, 'Execution finished without errors' without giving proper details.  My log gives more verbose details.

![image](https://user-images.githubusercontent.com/7872218/229375974-4915c245-e931-48c3-a4b7-c64984242171.png)

If you click 'View' and 'Sql Log' (if its not checked) to display the error log, then select 'user' in the dropdown, does it display more details?

![image](https://user-images.githubusercontent.com/7872218/229376037-0af7f348-3ae9-4dbc-a039-d24d4c1cb324.png)

I noticed you were using the sqlCipher version (which I don't normally) but can confirm the same results (for me) occurs in both.
I can't replicate your issue.

**Comment by bartek990** [comment]
I found something.
It has to do with something with comments.
```
-- Utworzenie tabeli
CREATE TABLE stock_market (
	ticker	TEXT NOT NULL,
	company	TEXT NOT NULL,
	price	TEXT NOT NULL
);

-- Wstawienie kilku wierszy
INSERT INTO stock_market (ticker, company, price)
VALUES	('CDR', 'CD Projekt', 260),
		('PLW', 'Playway', 650),
		('TEN', 'Ten Square Games', 550),
		('PCF', 'PCF Group', 80),
		('11B', '11 BIT Studios', 550),
		('BBT', 'Boombit', 18),
		('CFG', 'Creative Forge', 37);
		
SELECT * FROM stock_market;
SELECT *, ROWID FROM stock_market;

SELECT *, rowid FROM stock_market WHERE ROWID = 5;
SELECT * FROM stock_market WHERE ROWID = 5;

EXPLAIN QUERY PLAN SELECT * FROM stock_market WHERE ROWID = 5;
-- SEARCH TABLE stock_market USING INTEGER PRIMARY KEY (rowid=?)

SELECT * FROM stock_market WHERE ticker == 'BBT';
EXPLAIN QUERY PLAN SELECT * FROM stock_market WHERE ticker == 'BBT';
-- SCAN TABLE stock_market

DROP TABLE IF EXISTS stock_market;
```
![Zrzut ekranu z 2023-04-02 22-30-17](https://user-images.githubusercontent.com/6698414/229377480-bd3d2c89-6ade-440e-b933-12fb5833030f.png)

When I run this code with _F5_ everything works just fine, but when I run just `CREATE`, `INSERT` and `DROP` the table doesn't drop.
![Zrzut ekranu z 2023-04-02 22-35-41](https://user-images.githubusercontent.com/6698414/229377765-ae58d93e-524c-4d77-af12-5c9e8729bcee.png)

After I removed comments from code everything is working in both scenarios. Do I use some "illegal" words in comments? Should it work that way?

**Comment by chrisjlocke** [comment]
Can you create a new query selecting 1, then add a comment after it?

![image](https://user-images.githubusercontent.com/7872218/230785790-e422c163-8bab-4887-a2a6-3f2916a351bb.png)

Does it say 1 row returned?

I'm thinking this might be a Linux vs Windows thing.  I remember the parser is a bit finicky with carriage returns and getting the right line.

> but when I run just CREATE, INSERT and DROP the table doesn't drop.

How are you 'just' running those commands?
Highlighting a pressing F5, or highlighting and pressing 'run current line' (the button next to the execute button).
ie, >|, and not just >
Ensure its the former - highlight and press F5.

**Comment by bartek990** [comment]
> How are you 'just' running those commands?
Highlighting a pressing F5, or highlighting and pressing 'run current line' (the button next to the execute button).
ie, >|, and not just >
Ensure its the former - highlight and press F5.

I mean*Shift+F5* with cursor in code line.

There're no problems with `SELECT`.

**Comment by chrisjlocke** [comment]
So that makes sense then.
You're executing a comment, which is getting ignored.  Thus you're not getting the '1 rows executed'.  An ignoerd comment is nothing, so nothing is getting executed.
Highlight the SQL you want and press F5.

**Comment by bartek990** [comment]
Ok, but it's strange behaviour to run line above or below the line with cursor, don't you think?

And running whole script like in *read only* and *ignore errors* mode is also strange.

Maybe autofill should add `;` at the end of comments (although I cannot check if it works at the moment)?

Maybe app should highlight syntax in a different way?

Maybe 'run line' function shouldn't be executed if there is no selection?

For me this functionality is just broken. At least it should inform that something that is not a part of comment won't be executed. I remember that I started to use one line comments (`--`) because multiline comments `\**\` was behaving even stranger, but at least sql log was informing that only part of code was executed.

**Comment by chrisjlocke** [comment]
> Ok, but it's strange behaviour to run line above or below the line with cursor, don't you think?

No, but it's some convoluted logic working out where the cursor is, is the line above part of the same instruction, etc.
For example, I break up my SQL statement, so I normally have
select *
from widgets
where dateOrdered='2003-10-09'
and obsolete=0

If I'm in the middle and press shift+F5, DB4S knows not to just run 'from widgets' and uses the line above.  I think it even looks further down looking for a closing ; and gives up on a blank line.
But what if I enter
select *
from widgets
select *
from widgets

Two statements, so does it run both?  Is it the same statement?  We can eyeball it, but DB4S hasn't a scooby.  
Now this was coded (I believe) on a Mac (before it went all wacky with Linux) so this didn't play well with Linux line feeds (Windows and Linux use different line terminators).  
Many developers have looked at the code and tried to improve it.  We visit them often in hospital where they rock in the corner on their wrists, crying.
So in honour of their memories, we simply highlight the code and press F5.  It works a bit better.

> And running whole script like in read only and ignore errors mode is also strange

I don't follow .. I don't believe DB4S does this?

> Maybe 'run line' function shouldn't be executed if there is no selection?

As above .. that's its purpose.
It was originally designed for the developers who code their SQL in one line, eg,

select * from widgets where obsolete=0 and iveADirtyBasket=1 and myLeftFootTickles=true
select * from widgets where obsolete=10 and thisMakesNoSense='wibble' and myLeftHeadIsEmpty=true

etc.  Simply putting the cursor in a query and running 'in line' works.  Normally.

> For me this functionality is just broken

Which is why the 'highlight and press F5' works.  Honest, it does.

**Comment by bartek990** [comment]
Sorry, apparently my English expression skills are worse than I thought.

So... When I write script it works both with F5 and F5 + selection, but when I save it as a project file than it works with F5 (run all script) until I try to run one line of script with F5 + Selection. From that moment neither F5 nor F5 + selection runs any "writing command" (insert, drop etc) until I restart program.

I understand that part of code is abounded because of developers illness - sorry to hear that. Hope he get better.

"Bug Of Honour" is a strange label, but still better than saying everything is fine...

I see two solutions.
1. Cutting out comments before sending code to parser. It's crappy solution, but in these circumstances it fixes this bug. It should be simple to extract line that starts with '--' and ends with '\n' just before running.
2. Mark repository as 'legacy for Linux' or 'partial support for Linux '.

I was using this app as educational tool - without possibility of commenting and running parts of scripts it's useless for me so I'm using another software to manage SQL databases in my Python projects.
